pkg/util/errors/errors.go

```diff
@@ -4,17 +4,40 @@ import (
 	"net"
 	"net/url"
 	"os"
+	"strings"
 	"syscall"
 
 	"github.com/Azure/go-autorest/autorest"
 	"github.com/Azure/go-autorest/autorest/azure"
 )
 
+// some common network errors that we check for
+var str2Syscall = map[string]syscall.Errno{
+	"connection reset by peer":  syscall.ECONNRESET,
+	"connection refused":        syscall.ECONNREFUSED,
+	"network is unreachable":    syscall.ENETUNREACH,
+	"no such file or directory": syscall.ENOENT,
+}
+
+func unwrapNetHTTPError(err error) (bool, error) {
+	if !strings.HasPrefix(err.Error(), "net/http: HTTP/1.x transport connection broken:") {
+		return false, err
+	}
+	for sysMsg, errno := range str2Syscall {
+		if strings.Contains(err.Error()[47:], sysMsg) {
+			return true, errno
+		}
+	}
+	return false, err
+}
+
 // IsMatchingSyscallError returns true when the error is one of the Errno's in match
 // it deals with the different ways of wrapping the syscalls.
 func IsMatchingSyscallError(err error, match ...syscall.Errno) bool {
 	for {
 		switch t := err.(type) {
+		case nil:
+			return false
 		case *azure.RequestError:
 			err = &t.DetailedError
```

Comment:
I really think you can save one iteration of the for loop in the case where this function is called with a `*azure.RequestError` by doing:

```suggestion
			err = t.DetailedError.Original
```